### PR TITLE
Add Redis compatibility test to CI

### PR DESCRIPTION
**Pull request by bjosv**

### CODE REVIEW

File: test/ered_SUITE.erl [hunk 1]
```diff
@@ -70,8 +69,15 @@ wait_for_consistent_cluster() ->
                                 ered_client:stop(Pid),
                                 SlotMap
                         end(P) || P <- ?PORTS],
+            %% Get the amount of nodes in the first reply. The reply format has improved between
+            %% Redis 6 and 7, but to handle Redis 6 we count each expected port in the reply.
+            NoOfNodes = length([X || X <- [lists:member(integer_to_binary(P),
+                                                        binary:split(list_to_binary(hd(AllNodes)), <<"\n">>, [global]))
+                                           || P <- ?PORTS],
+                                     X =:= true]),
+
```
Comment (bjosv) [inline test/ered_SUITE.erl:78]: Is there a simpler way? (probably..)
Redis7 divides nodes with `\n\n` and then a `\n` between each info element for the node,
Redis 6 is just single newlines.

Comment (zuiderkwast) [reply test/ered_SUITE.erl:78]: What? THis is CLUSTER SLOTS, it should be a nested list without any newlines.

I see now that an ered_client connection is established, then it is not used but instead redis-cli is used via os:cmd to call CLUSTER SLOTS. Then the ered_client connection is closed. Obviously a mistake but if all we were interested in was that all nodes return identical information, this seemed to have worked..

File: test/ered_SUITE.erl [hunk 2]
```diff
@@ -67,8 +69,15 @@ wait_for_consistent_cluster() ->
...
             case length(lists:usort(AllNodes)) of
-                1 ->
+                1 when NoOfNodes == length(?PORTS) ->
                     true;
                 _ when N > 0 ->
                     timer:sleep(500),
```
Comment (zuiderkwast) [inline test/ered_SUITE.erl:83]: Here, if we would want to check the contents of the CLUSTER SLOTS output, after checking that they're all identical, we could do

```Erlang
            case lists:usort(AllNodes) of
                [Nodes] ->
                    %% ... Check the contents of Nodes here ...
                    true;
                _NotAllIdentical ->
                    timer:sleep(500),
```

File: test/ered_SUITE.erl (hunk 1 above)
Comment (bjosv) [reply test/ered_SUITE.erl]: Yikes.. I must have got temporary code blindness.. missed that. Thanks

File: test/ered_SUITE.erl [hunk 3]
```diff
@@ -59,30 +59,30 @@ init_per_suite(_Config) ->
...
-                                                        binary:split(list_to_binary(hd(AllNodes)), <<"\n">>, [global]))
-                                           || P <- ?PORTS],
-                                     X =:= true]),
-
-            case length(lists:usort(AllNodes)) of
-                1 when NoOfNodes == length(?PORTS) ->
-                    true;
-                _ when N > 0 ->
+            case lists:usort(AllNodes) of
+                [Node] ->
```
Comment (zuiderkwast) [inline test/ered_SUITE.erl:72]: Perhaps confusing variable name. It's not a node. It's the slotmap response from all nodes, when they're all identical.

```suggestion
                [Slotmap] ->
```

File: test/ered_SUITE.erl [hunk 4]
```diff
@@ -59,30 +59,30 @@ init_per_suite(_Config) ->
...
+                            Loop(N-1);
+                        false ->
+                            error({timeout_consistent_cluster, Node, ?PORTS})
+                    end;
+                _NotAllIdentical when N > 0 ->
                     timer:sleep(500),
                     Loop(N-1);
-                _ ->
+                _NotAllIdentical ->
                     error({timeout_consistent_cluster, AllNodes})
```
Comment (zuiderkwast) [inline test/ered_SUITE.erl:86]: This is duplicated logic. How about returning a success boolean from the `case` expression and then afterwards add another case expression for handling it (ok, loop or error)?

**Review by zuiderkwast — APPROVED** [review]
LGTM

**Pull request merged**